stats/bridge_stats.go

```diff
@@ -228,14 +229,41 @@ func (s *Service) login(conn *connWrapper) error {
 	return nil
 }
 
+// reportLoop loops as long as the connection is alive and try to report the bridge stats
+// every sendStatsTicket. If the conenction is drop or we got any issues, we will exist the
+// function for reprocessing the flow
+func (s *Service) reportLoop(conn *connWrapper) error {
+	var err error
+
+	sendStatsTicker := time.NewTicker(5 * time.Second) // Every 5 seconds
+
+	for err == nil {
+		select {
+		case <-s.quitCh:
+			sendStatsTicker.Stop()
+			conn.Close()
+			return nil
+		case msg := <-s.errCh:
+			err = s.setLastError(msg.Listener, msg.Err)
+		case msg := <-s.processedBlockCh:
+			err = s.setProcessedBlock(msg.Listener, msg.ProcessedBlock)
+		case <-sendStatsTicker.C: // Checking stats interval
+			if err = s.report(conn); err != nil {
+				log.Warn("bridge stats report failed", "err", err)
+				// When bridge stats report failed need to relogn after 10 seconds
+			}
+		}
+	}
+	sendStatsTicker.Stop()
+	return nil
+}
+
 func (s *Service) report(conn *connWrapper) error {
-	s.lock.Lock()
-	defer s.lock.Unlock()
 	// collect bridge stats and send to ronin stats
 	info := &BridgeInfo{
 		Node:           s.node,
 		Operator:       s.operator,
-		Version:        s.version,
+		Version:        "v2",
```

Comment:
Please make this a constant for easy future update

Reply:
So version in Service struct is unused, we need to remove that field and replace where that field is used with this constant too

Reply:
@minh-bq  Fixed